src/Index/Index.php

```diff
@@ -300,23 +306,61 @@ function( ?array $carry, SchemaField $item = null ) : array {
     }
 
     /**
-     * Index all posts to the RediSearch database
+     * Create a wp cron event chain to index all posts.
      *
-     * @param  \WP_REST_Request|null $request Rest request details or null if not rest api request.
-     * @return int                            Amount of items indexed.
+     * @param  int|WP_Rest_Request $args Int for offset or WP_Rest_Request on first run.
+     * @return true|false|WP_Error       Result of next wp_schedule_single_event call or true on final run.
      */
-    public function index_all( \WP_REST_Request $request = null, array $query_args = [] ) : int {
+    public function schedule_partial_index( $offset = null ) {
+
+        // If this was created by a user via the admin just schedule without running the actual import
+        if ( $offset instanceof \WP_REST_Request ) {
+
+            // Make sure we don't create new cron jobs if one is already running
+            $cron = \get_option( 'cron' );
+            foreach ( $cron as $timestamp => $events ) {
+                foreach ( $events as $hook => $args ) {
+                    if ( $hook === static::HOOKS['schedule_partial_index'] ) {
+                        return true;
+                    }
+                }
+            }
+
+            $offset = $offset->get_param( 'offset' ) ?? 0;
+            return \wp_schedule_single_event( time(), static::HOOKS['schedule_partial_index'], [ $offset ], true );
+        }
+
+        // Run import
+        $offset = \is_int( $offset ) ? $offset : 0;
+        $count  = $this->index_all([
+            'limit'  => \apply_filters( static::HOOKS['schedule_partial_index_limit'], 400 ),
```

Comment:
The 400 default value here is just taken by testing on the sbl project local development with the cron running every 1min and it didnt seem to cause problems.
If running the import via wp cli it took about 10sec for it to import 400 posts.